ff/serving/metadata/metadata_test.go

```diff
@@ -0,0 +1,233 @@
+	}
+	listAll, err := lookup.List()
+	if err != nil {
+		t.Fatalf("Failed to list all resources for the 1 resource case %s", err)
+	}
+	if len(listAll) != 1 {
+		t.Fatalf("Failed to get the correct number of results = 1")
+	}
+	var resourceType ResourceType = "CSV"
+	listByType, errListForType := lookup.ListForType(resourceType)
+	if errListForType != nil {
+		t.Fatalf("Failed to list resources (by type) for the 1 resource case %s", errListForType)
+	}
+	if len(listByType) != 1 {
+		t.Fatalf("Failed to get the correct number of results (1)")
+	}
+
+}
+
+func TestMethodsThreeResource(t *testing.T) {
+	lookup := make(localResourceLookup)
+	testIds := []ResourceID{
+		{
+			Name:    "Fraud",
+			Variant: "default",
+			Type:    ResourceType("CSV"),
+		}, {
+			Name:    "Time",
+			Variant: "second",
+			Type:    ResourceType("Feature"),
+		}, {
+			Name:    "Ice-Cream",
+			Variant: "third-variant",
+			Type:    ResourceType("user"),
+		},
+	}
+	for _, resource := range testIds {
+		if err := lookup.Set(resource, &mockResource{id: resource}); err != nil {
+			t.Fatalf("Failed to Set %s", err)
+		}
+	}
+	if _, err := lookup.Lookup(testIds[0]); err != nil {
+		t.Fatalf("Failed to return a Lookup error because %s", err)
+	}
+	containsId, err := lookup.Has(testIds[1])
+	if err != nil {
+		t.Fatalf("Failed to run Has function %s", err)
+	}
+	if containsId == false {
+		t.Fatalf("Failed to return true in Has function for the 3 resource case")
+	}
+	listAll, err := lookup.List()
+	if err != nil {
+		t.Fatalf("Failed to list all resources for the 3 resource case %s", err)
+	}
+	if len(listAll) != 3 {
+		t.Fatalf("Failed to get the correct number of results = 3")
+	}
+	var resourceType ResourceType = "CSV"
+	listByType, errListForType := lookup.ListForType(resourceType)
+	if errListForType != nil {
+		t.Fatalf("Failed to list resources (by type) for the 1 resource case %s", errListForType)
+	}
+	if len(listByType) != 1 {
+		t.Fatalf("Failed to get the correct number of results (1)")
+	}
+	res, errSubmap := lookup.Submap(testIds)
+	if errSubmap != nil {
+		t.Fatalf("Failed to create a submap %s", errSubmap)
+	}
+	listsubmap, errList := res.List()
+	if errList != nil {
+		t.Fatalf("Failed to get list of submap %s", errList)
+	}
+	if len(listsubmap) != 3 {
+		t.Fatalf("Failed to return a corrects submap")
+	}
+	resNotContain, errSubmapNotContain := lookup.Submap([]ResourceID{
+		{
+			Name:    "Banking",
+			Variant: "Secondary",
+			Type:    ResourceType("provider"),
+		},
+	})
+	if errSubmapNotContain == nil && resNotContain != nil {
+		t.Fatalf("Failed to catch error not found in submap %s", errSubmap)
+	}
+}
+
+func TestMethodsTypesense(t *testing.T) {
+	sugar := zap.NewExample().Sugar()
+	server := Config{
+		Logger: sugar,
+		TypeSenseParams: &search.TypeSenseParams{
+			Host:   "localhost",
+			Port:   "8108",
+			ApiKey: "xyz",
+		},
+	}
+	testIds := []ResourceID{
+		{
+			Name:    "Fraud",
+			Variant: "default",
+			Type:    ResourceType("CSV"),
```

Comment:
there is a list of ResourceType s toward the top of metadata.go that you should use. It's in a const ( ) block